### PR TITLE
reciprocal and Domain retention

### PR DESCRIPTION
**Pull request by kgaonkar6**
checkReciprocal creates column `reciprocal_exists` in annotate_fusion_calls()

checkDomainStatus creates columns "DomainRetainedInGene1A" ,"DomainRetainedInGene1B" here the domainsToCheck will be given a character list of pfamIDs in fusion_driver() since it seems to be more of a project specific criteria, does that sound ok?
 
`kinaseid<-unique(bioMartDataPfam$pfam_id[grep("kinase",bioMartDataPfam$NAME)] )`  

the code will check status of the domain in either Gene1A and Gene1B

### CODE REVIEW

File: R/annotate_fusion_calls.R [hunk 1]
```diff
@@ -3,10 +3,11 @@
 #' @param standardFusioncalls A dataframe from star fusion or arriba standardized to run through the filtering steps
 #' @param geneListReferenceDataTab A dataframe with column 1 as GeneName 2 source file 3 type; collapse to summarize type
 #' @param fusionReferenceDataTab A dataframe with column 1 as FusionName 2 source file 3 type; collapse to summarize type
-#'
+#' @param checkReciprocal Logical value to check if fusion also has reciprocal fusion in Sample, default to FALSE
```
Comment (jharenza) [inline R/annotate_fusion_calls.R]:
```suggestion
#' @param checkReciprocal Logical value to check if fusion also has reciprocal fusion in Sample, default to TRUE
```

@kgaonkar6 I think this should be a default == T

File: R/fusion_driver.R [hunk 2]
```diff
@@ -7,10 +7,13 @@
 #' @param annotated Boolean if annotated
 #' @param geneListReferenceDataTab A dataframe with column 1 as GeneName 2 source file 3 type; collapse to summarize type
 #' @param fusionReferenceDataTab A dataframe with column 1 as FusionName 2 source file 3 type; collapse to summarize type
+#' @param checkDomainStatus Logical value to check if domain status in fused gene for given domansToCheck, default to FALSE
```
Comment (jharenza) [inline R/fusion_driver.R:12]:
```suggestion
#' @param checkDomainStatus Logical value to check if domain status in fused gene for given domainsToCheck, default to TRUE
```

also think this should be default == T

**Comment by kgaonkar6** [comment]
the `checkDomainStatus` adds columns "DomainRetainedInGene1A" ,"DomainRetainedInGene1B" by checking for overlap with `domainsToCheck` ( we are using kinase pfamIDS). If we set it by default = TRUE do we always check for kinase domain by default as well? I wasn't sure if we want to always check for kinase domians so made it FALSE by deafult

**Comment by kgaonkar6** [comment]
Also adding the inst/extdata and demo file #66 and updating `Putative driver filtering` vignette

**Review by jharenza — APPROVED** [review]

**Pull request merged**